nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/StreamWorkManager.java

```diff
@@ -169,7 +170,11 @@ public void schedule(Work work, Scheduling scheduling, boolean afterCommit) {
             return;
         }
         String key = work.getPartitionKey();
-        appender.append(key, Record.of(key, WorkComputation.serialize(work)));
+        LogOffset offset = appender.append(key, Record.of(key, WorkComputation.serialize(work)));
+        // we store offset to handle coalescing works or state storage when several works with same id are scheduled
+        if (work.isCoalescing() || storeState) {
```

Comment:
|| not && ?

Reply:
It's not, we store the offset if the work is coalescing or if we want to storeState

Reply:
There is no more reason to update offset when storeState is true, also remove comment line 174